### DISCUSSION TITLE
Updating reference database snapshot

### DISCUSSION THREAD

**Discussion by mflynn-lanl** [microbiomedata/nmdc-edge#272] [General]
@edwins Every so often the reference databases change and I'd like to automate the update of the snapshot as much as possible. To do this, I would like to mount a volume created from the old snapshot, update the files, detach the volume and create a new snapshot. I used these instructions when I created the original:
```
sudo pvcreate /dev/sdc
sudo vgcreate project_vg /dev/sdc
sudo lvcreate -l+100%VG -n project project_vg
sudo mkfs.xfs -L project /dev/project_vg/project
sudo mount /dev/project_vg/project /mnt/
```
How do I adapt them for updating volume I use for the snapshot? When I attach the volume to the VM in Horizon, it says it is attached to `/dev/sdc` so I don't need to run `pvcreate`, right? What about the rest? I want to make sure I don't cause more problems for myself later.

**Comment by edwins** [comment]
for existing lvm, i believe you just need to mount it -- you shouldn't be doing any create operations and definitely do not run mkfs again.

**Reply by mflynn-lanl** [reply]
```
exouser@ref-db:~$ sudo mount /dev/sdd /mnt
mount: /mnt: unknown filesystem type 'LVM2_member'.
```

**Reply by mflynn-lanl** [reply]
I'm now running into problems due to my reference database on the volume snapshot being out of date. Can you help me figure out why I'm not able to mount a volume created from the snapshot to the VM with the reference data?

**Reply by edwins** [reply]
I'm on vacation until Tuesday. Are you able to mount this volume in a new vm or is it just magic cluster vms? If it's a new vm, you can post an issue into js2 for help more immediately.

**Reply by mflynn-lanl** [reply]
This is for the MC VMs. I can mount it to some of my non-MC VMs but not others

**Comment by mflynn-lanl** [comment]
I'm not able to mount the reference data volume after I've unmounted it, detached it from the VM instance and taken a snapshot. When I try to mount the volume the second time I don't see `/dev/project_vg/project` when I run `sudo fdisk -l` what I do see is:
```
Disk /dev/sdc: 2.93 TiB, 3221225472000 bytes, 6291456000 sectors
Disk model: QEMU HARDDISK
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
```

I've been getting around this by creating new VMs each time I want to update the reference data volume snapshot

**Reply by edwins** [reply]
I can look into making a cacao template to help you manage more easily, but probably won't be able to start that until next week.
